From 09f686b243c57ea9e9c9c126da8805d1257d0f6d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:41:07 +0800
Subject: [PATCH] Revert "Bluetooth: Do not attempt to send dlci disconnect
 when in BT_CONFIG."

This reverts commit 8a4424765e8daaf56a2d8a0e0ab9dfcf859879fa.
---
 net/bluetooth/rfcomm/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/bluetooth/rfcomm/core.c b/net/bluetooth/rfcomm/core.c
index f224b2db0249..25692bc0a342 100755
--- a/net/bluetooth/rfcomm/core.c
+++ b/net/bluetooth/rfcomm/core.c
@@ -459,6 +459,7 @@ static int __rfcomm_dlc_close(struct rfcomm_dlc *d, int err)
 
 	switch (d->state) {
 	case BT_CONNECT:
+	case BT_CONFIG:
 		if (test_and_clear_bit(RFCOMM_DEFER_SETUP, &d->flags)) {
 			set_bit(RFCOMM_AUTH_REJECT, &d->flags);
 			rfcomm_schedule(RFCOMM_SCHED_AUTH);
-- 
2.35.3

